src/pages/components/controls/filenameControls.tsx

```diff
@@ -30,3 +32,34 @@ export const PatternToken = memo(
     )
   }
 )
+
+type SortablePatternTokenProps = {
+  tokenName: string
+  innerRef: (element?: HTMLElement) => void
+  isDragging: boolean
+  draggableProvided: DraggableProvided
+}
+
+export const SortablePatternToken = memo(
+  ({ tokenName, innerRef, isDragging, draggableProvided }: SortablePatternTokenProps) => {
+    return (
+      <Flex
+        {...draggableProvided.draggableProps}
+        ref={innerRef}
+        gap={0.5}
+        paddingInline={'0.5em'}
+        fontSize={'1em'}
+        fontWeight={'medium'}
+        color={'gray.900'}
+        borderRadius={'full'}
+        bg={'white'}
+        opacity={isDragging ? 0.7 : 1}
+      >
+        <Center {...draggableProvided.dragHandleProps}>
+          <DragHandleIcon boxSize={3} color={'gray.600'} />
+        </Center>
+        {tokenName}
+      </Flex>
+    )
```

Comment:
Works fine.  Just some UX consideration.
+ If the whole token can be draggle that would be good.
+ Change cursor style by dragging status, `cursor={isDragging ? 'grabbing' : 'grab'}`.
+ Prevent text selection, `userSelect={'none'}`.
+ Token with `X` icon for removing from the slot. ([Ref.](https://chakra-ui.com/docs/components/tag/usage#with-close-button))